devchat.go

```diff
@@ -431,6 +457,77 @@ func (u *user) displayPronouns() string {
 	return result[1:]
 }
 
+func (u *user) savePrefs() error {
+	export := userexport{}
+
+	export.Name = stripansi.Strip(u.name)
+	export.Pronouns = u.pronouns
+	export.Bell = u.bell
+	export.PingEverytime = u.pingEverytime
+	export.FormatTime24 = u.formatTime24
+	export.Color = u.color
+	export.ColorBG = u.colorBG
+	export.Timezone = u.timezone
+	export.Autoload = u.autoload
+
+	saved, err := json.Marshal(export)
+	if err != nil {
+		return err
+	}
+	xdg_data_dir := os.Getenv("XDG_DATA_DIR")
+
+	if xdg_data_dir == "" {
+		// According to XDG, this is the default if it's unset
+		xdg_data_dir = filepath.Join(os.Getenv("HOME"), ".local", "share")
+	}
+
+	save := filepath.Join(xdg_data_dir, "/devzat")
+	os.Mkdir(save, 0755)
+	save = filepath.Join(save, u.id)
+
+	f, err := os.Create(save)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	f.Write(saved)
+	return nil
+}
+
+func (u *user) loadPrefs(firsttime bool) error {
+	xdg_data_dir := os.Getenv("XDG_DATA_DIR")
+	if xdg_data_dir == "" {
+		xdg_data_dir = filepath.Join(os.Getenv("HOME"), ".local", "share")
+	}
+
+	save := filepath.Join(xdg_data_dir, "/devzat", u.id)
+
+	contents, err := ioutil.ReadFile(save)
+	if err != nil {
+		return err
+	}
+
+	export := userexport{}
+	json.Unmarshal(contents, &export)
+
+	if !export.Autoload && firsttime {
```

Comment:
Won't `export.Autoload` always be true since prefs are saved only if the user enables autoload?

Reply:
No, because of save and load commands

Reply:
I'm trying to understand what `firsttime`'s purpose is: could you explain it to me?